VideoLocker/src/main/java/org/edx/mobile/view/CourseDashboardFragment.java

```diff
@@ -30,111 +28,124 @@
+
+            holder.typeView.setIcon(Iconify.IconValue.fa_comments_o);
+            holder.titleView.setText(R.string.discussion_title);
+            holder.subtitleView.setText(R.string.discussion_subtitle);
+            holder.rowView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    environment.getRouter().showCourseDiscussionTopics(getActivity(), courseData);
+                }
+            });
+
+            holder = createViewHolder(inflater, parent);
+
+            holder.typeView.setIcon(Iconify.IconValue.fa_file_text_o);
+            holder.titleView.setText(R.string.handouts_title);
+            holder.subtitleView.setText(R.string.handouts_subtitle);
+            holder.rowView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if (courseData != null)
+                        environment.getRouter().showHandouts(getActivity(), courseData);
+                }
+            });
+
+            holder = createViewHolder(inflater, parent);
+
+            holder.typeView.setIcon(Iconify.IconValue.fa_bullhorn);
+            holder.titleView.setText(R.string.announcement_title);
+            holder.subtitleView.setText(R.string.announcement_subtitle);
+            holder.rowView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    if (courseData != null)
+                        environment.getRouter().showCourseAnnouncement(getActivity(), courseData);
+                }
+            });
+        } else {
+            errorText.setText(courseData.getCourse().getCoursewareAccess().getUser_message());
+        }
     }
 
     @Override
     public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
-        try {
-            final Bundle bundle = getArguments();
-            courseData = (EnrolledCoursesResponse) bundle
-                    .getSerializable(CourseData);
-
-            if ( courseData == null )
-                return;
-
-            final String headerImageUrl = courseData.getCourse().getCourse_image(environment.getConfig());
-            Glide.with(CourseDashboardFragment.this)
-                    .load(headerImageUrl)
-                    .placeholder(R.drawable.edx_map_login)
-                    .transform(new TopAnchorFillWidthTransformation(getActivity()))
-                    .into(headerImageView);
-
-            courseTextName.setText(courseData.getCourse().getName());
-            CourseEntry course = courseData.getCourse();
-            courseTextDetails.setText(course.getDescription(getActivity()));
-
-        } catch (Exception ex) {
-            logger.error(ex);
-        }
+
+        if (courseData == null || !IS_COURSEWARE_ACCESSIBLE)
+            return;
```

Comment:
This should either be on the same line as the `if` condition, or it should be inside brackets.